lib/formats/uri/url_utils.flow

```diff
@@ -0,0 +1,207 @@
+import formats/html/html;
+import net/appurl;
+import string;
+import sys/target;
+
+
+export {
+	// Does the url contains `.php` substring?
+	isPhpUrl(url : string) -> bool;
+
+	// Determines if a given URL is considered a valid PHP URL (with additional checks).
+	isPhpUrlImproved(url : string) -> bool;
+	
+	// Splits a URL into its body and query parameters, returning them as a pair.
+	splitUrlToParams(url : string) -> Pair<string, [string]>;
+
+	// Extracts the local filename from a URL (if presented in the URL).
+	getLocalFileName(url : string) -> string;
+
+	// Removes a specified parameter from a URL, handling only PHP-style URLs.
+	removeTokenFromUrl(url : string) -> string;
+
+	// Removes a JWT token from a URL, handling only PHP-style URLs.
+	removeParameterFromUrl(url : string, key : string) -> string;
+
+	// Adds a JWT token from a URL, handling only PHP-style URLs.
+	addNewTokenToUrl(jwt : string, url : string) -> string;
+
+	// Adds or replaces a JWT token in a URL, handling only PHP-style URLs.
+	addOrReplaceTokenToUrl(jwt : string, url : string) -> string;
+
+	// Adds a query parameter `ignoreRangeHeaders=1` to the given URL
+	//  if the user is using a Safari browser and
+	//  the applicaeion runs with the parameter `TestSafari=true`.
+	addNoRangeForSafari(url : string) -> string;
+
+	// Extracts the value of a specific query parameter from a URL.
+	extractUrlParamValue(url : string, key : string) -> string;
+
+	// Replaces the value of a specific query parameter from a URL.
+	replaceUrlParamValue(url : string, key : string, value : string) -> string;
+
+	// Filters the query parameters of a URL, retaining only those that are relevant
+	//  for caching purposes within a PHP context.
+	cropUrlToCheckInCache(url : string) -> string;
+
+	// Temporary disable RequestHeaders processing.
+	// For instance to keep url in media src attribute instead of blob object.
+	setUnsetRequestHeadersLess() -> () -> void;
+
+	// Register a new converter to create a request header from a token in a URL.
+	setTokenToHeaderConverter(tokenName : string, converter : (string) -> KeyValue) -> void;
+
+	// Split URL string into a string without tokens and request headers with tokens.
+	splitOnUrlAndHeaders(url : string) -> Pair<string /*url*/, [KeyValue] /*headers*/>;
+
+	// The same as above but with callback implementation.
+	splitOnUrlAndHeadersWithCallback(mediaUrl : string, callback : (url : string, headers : [KeyValue]) -> ?) -> ?;
+}
+
+requestHeadersLessR = ref false;
+setUnsetRequestHeadersLess() -> () -> void {
+	requestHeadersLessR := true;
+	\-> requestHeadersLessR := false;
+}
+
+isPhpUrl(url : string) -> bool {
+	strContains(url, ".php?");
+}
+
+isPhpUrlImproved(url : string) -> bool {
```

Comment:
I understand that you just moved this file from innovation repo, but this fn naming looks strange. It doesn't look like it really check for php